### ISSUE TITLE
Godot Shader Language Support?

### ISSUE THREAD

**Issue by hayahane** [JetBrains/godot-support#124]
Currently there is no support for godot shader file.
Will there be basic support like highlighting and code completion?

**Comment by van800** [comment]
I think there is highlighting in the 2023.3.x.

**Comment by van800** [comment]
Please attach your shader file for me to check.

**Comment by hayahane** [comment]
```glsl
shader_type spatial;
render_mode cull_back;

#include "shader_include/color_convert.gdshaderinc"

global uniform float threshold;

void light() {
	float dot_result = dot(NORMAL,LIGHT);
	if (LIGHT_IS_DIRECTIONAL)
	{
		DIFFUSE_LIGHT += step(threshold, dot_result) * ATTENUATION * LIGHT_COLOR;
	}
	else
	{
		float v = rgb_to_hsv(DIFFUSE_LIGHT).z;
		vec3 color = DIFFUSE_LIGHT + ATTENUATION * LIGHT_COLOR;
		color = vec3(rgb_to_hsv(color).rg,v);
		DIFFUSE_LIGHT = hsv_to_rgb(color);
	}
}
```
Only found shader_type working.
![屏幕截图 2024-01-08 151206](https://github.com/JetBrains/godot-support/assets/39263675/cf64047a-f025-4111-aa02-0b533979a806)

**Comment by van800** [comment]
Got it. Thank you for the report.

We may start maintaining our own textmate bundle, combining
https://github.com/godotengine/godot-vscode-plugin (for gdscript), which we use now
and this other one https://github.com/AlfishSoftware/godot-files-vscode, which has a better gdshader support.
Both have permissive licenses.

**Comment by van800** [comment]
For now you may checkout the https://github.com/AlfishSoftware/godot-files-vscode anywhere on your drive and use it.
Open Rider settings, type TextMate there, add new bundle from the checkout folder / "extension", 
and also uncheck the "quequlim/godot-tools".
Unfortunately this way you'd loose the gdscript highlighting, and get the gdshader.
I have reported the problem to corresponding devs in the IntelliJ Platform, however can't promise fast resolution.
I hope to come up with some alternative solution.

**Comment by van800** [comment]
Got a [fix](https://github.com/JetBrains/godot-support/commit/00cbb0663f28bd0dc097a2c056c43e141d8a0236) for that. Would be available with 2024.1 EAP1 at some point in January.

**Comment by Insofan** [comment]
gdshader not work on rider 2025 c# project
